src/test/regress/sql/gporca.sql

```diff
@@ -3341,6 +3341,74 @@ with cte as (
 select * 
 from empty_cte_tl_test
 where id in(select id from cte);
+
+-- check that query which hangs before patch https://github.com/greenplum-db/gpdb/pull/14896 (1),
+-- and fallbacks to postgres due to patch https://github.com/arenadata/gpdb/pull/302 above (1),
+-- now doesn't fallback
+
+-- start_matchsubs
+-- m/Executor [mM]emory: .*/
+-- s/Executor [mM]emory: .*/Executor Memory: /
+-- m/Extra Text:.*/
+-- s/Extra Text:.*/Extra Text:/
+-- m/Memory used:\s+[0-9]+kB.*/
+-- s/Memory used:\s+[0-9]+kB.*/Memory used:/
+-- m/\(actual time=\d+\.\d+..\d+\.\d+ rows=\d+ loops=\d+\)/
+-- s/\(actual time=\d+\.\d+..\d+\.\d+ rows=\d+ loops=\d+\)/(actual time=##.###..##.### rows=# loops=#)/
+-- m/Execution time: \d+\.\d+ ms/
+-- s/Execution time: \d+\.\d+ ms/Execution time: ##.### ms/
+-- m/Planning time: \d+\.\d+ ms/
+-- s/Planning time: \d+\.\d+ ms/Planning time: ##.### ms/
+-- end_matchsubs
+
+DROP TABLE tab1, tab2;
+CREATE TABLE tab1(dict_id INT, tab_1_join_col INT);
+CREATE TABLE tab2(part_column INT, tab_1_join_col INT);
+CREATE TABLE dict_tab(dict_id INT PRIMARY KEY) DISTRIBUTED REPLICATED;
+
+EXPLAIN (ANALYZE, COSTS OFF) WITH
```

Comment:
There is more simple reproduction
```sql
CREATE TABLE d (a int, b int, c int) DISTRIBUTED BY (a);
CREATE TABLE r (a int, b int, c char(255)) DISTRIBUTED REPLICATED;
INSERT INTO d SELECT 1, generate_series(1,10), 1;
INSERT INTO r SELECT 1, 2, generate_series(1,100);
EXPLAIN (ANALYZE off, COSTS off, VERBOSE off)
WITH e AS (
    SELECT DISTINCT b FROM d
) SELECT * FROM r JOIN e USING (b) JOIN e f USING (b);
```

Reply:
This query doesn't hangs on the gpdb 6.22.41 (seems due to the first verison of ADBDEV-3098 patch), and there is no false-positive fallback into postgres optimizer.

Reply:
on current dev it yuilds
```sql
2023-07-12 15:52:31:819837 +05,THD000,ERROR,"Feature not supported: CTE Consumer without the appropriate CTE Producer under a duplicate-hazard motion or a tainted replicated node",
2023-07-12 15:52:31:821084 +05,THD000,ERROR,"Feature not supported: CTE Consumer without the appropriate CTE Producer under a duplicate-hazard motion or a tainted replicated node",
psql:/home/ADBDEV/3888/CTE1.sql:34: LOG:  Pivotal Optimizer (GPORCA) failed to produce plan (unexpected)
```

Reply:
As I understand, the dev build contains the final version of your patch (ADBDEV-3168) from upstream. And the plan which was generated for this query, is considered as hazard (hasUnpairedCTEConsumers makes a faulse-positive decision) as a result - fallback

Reply:
yes, by this, reproduction is significant simpler

Reply:
on the 6.22.41 without patch 3098, the query cause a segfault
```sql
EXPLAIN (ANALYZE off, COSTS off, VERBOSE off)
WITH e AS (
    SELECT DISTINCT b FROM d
) SELECT * FROM r JOIN e USING (b) JOIN e f USING (b);
```

As I understand the suggestion is to change case with hanging to simplified repro and, as a result the second case becomes useless - so only one test case is needed. May be one case would be enough

Reply:
Now the description of the pull request does not match the test.